### PR TITLE
Use buffer reading functions

### PR DESCRIPTION
**Pull request by spalmer25**
This PR use function on buffer.
This makes the code clearer and closer to what is done in [LedgerHQ/app-boilerplate](https://github.com/LedgerHQ/app-boilerplate)

It prepares issue #74 

Depends on #79

### CODE REVIEW

**Comment by ajinkyaraj-23** [comment]
Can you rebase on main?  it will be easier to review.

File: src/operations.h [hunk 1]
```diff
@@ -98,6 +108,9 @@ struct nexttype_subparser_state {
 
         struct delegation_contents dc;  ///< wire delegation content
 
+        // Required to read Reveal public key
+        union public_key pk;  ///< wire public key
```
Comment (ajinkyaraj-23) [inline src/operations.h:112]: This is not yet being used right?

**Comment by ajinkyaraj-23** [comment]
Also if its not draft stage, please change the target branch to main

File: src/operations.h (hunk 1 above)
Comment (spalmer25) [reply src/operations.h:112]: Yes, this field and the others above it are not used, they are only needed for `nexttype_subparser` to work properly.

**Pull request merged**